corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -51,21 +51,28 @@ hqDefine("data_dictionary/js/data_dictionary", [
                     var propObj = propertyListItem(prop.name, prop.label, false, group.name, self.name, prop.data_type,
                         prop.description, prop.allowed_values, prop.fhir_resource_prop_path, prop.deprecated,
                         prop.removeFHIRResourcePropertyPath, isGeoCaseProp, prop.is_safe_to_delete, prop.id);
-                    propObj.description.subscribe(changeSaveButton);
-                    propObj.label.subscribe(changeSaveButton);
-                    propObj.fhirResourcePropPath.subscribe(changeSaveButton);
-                    propObj.dataType.subscribe(changeSaveButton);
-                    propObj.deprecated.subscribe(changeSaveButton);
-                    propObj.deleted.subscribe(changeSaveButton);
-                    propObj.removeFHIRResourcePropertyPath.subscribe(changeSaveButton);
+                    subscribePropObservable(propObj, propObj.description, changeSaveButton);
```

Comment:
I was going to ask if there was evidence that this would be a noticeable performance improvement - sometimes the additional complexity added by performance enhancements isn't worth it - but seeing that you can hook into existing code, this seems reasonable.

Reply:
I think the most noticable improvements will be when there are a lot of case properties (e.g. 500). Instead of having to do 500 DB calls to retrieve each case property and then 500 more to save them, we now only have to do DB calls for properties that have actually changed. Being able to plug into the existing code is definitely another big plus.

Reply:
Yeah, I'm in agreement that it's a theoretical improvement. But when there are 500 case properties, does it shave a few seconds off the save, or ten minutes? And how often does someone save 500 case properties? Then those answers should be weighed against the additional code complexity and maintenance cost.

But given that the additional complexity turned out to be pretty minor, I'm in favor of this change even not knowing the answers to the questions about actual performance gain.

Reply:
Out of curiousity, I did some performance timing testing on my local environment (grated timings would be different on prod). I created a Data Dictionary with 500 case properties and for testing only updated one case property. The timings were as follows:
- Without changes from this PR: Avg. 3.9 seconds
- With changes from this PR: Avg. 0.15 seconds

I imagine this difference would go up as the number of case properties increase, but generally it seems like the time shaved is in the ballpark of seconds.

Reply:
Interesting, that's great info to have!

Reply:
that is great info, I like facts! 
thanks @zandre-eng

Reply:
Just for some perspective, we have 15 domains on prod with over 10,000 case properties across all case types. One domain has 44.6K. And we have 21 domains with an average of more than 500 case properties per case type, 3 have more than 1000 per case type. So these performance improvements seem helpful.